hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -637,6 +635,8 @@ def generate_new_examples(self):
         if Phase.generate not in self.settings.phases:
             return
 
+        self.health_check_state = HealthCheckState()
```

Comment:
Moving this is a change in behaviour that doesn't seem to be mentioned anywhere in the `RELEASE.rst` (and is sufficiently unrelated to the already rather large rest of the pull request that I'd rather it be in a separate one)